Merge options and file.data when both are passed

kt3k opened the pull request:
Hello

Thanks for the plugin.

In my team's use case, we'd like to pass both file.data and options. This PR adds the feature of merging `options` and `file.data` if both are given. I think this is a natural enhancement.

rogeriopvl commented on test/helper.js:
```diff
@@ -0,0 +1,8 @@
+ */
+exports.removeLineBreak = function (contents) {
+    return contents.toString().replace(/\r?\n|\r/g, '');
+};
```
Why is this function needed?

kt3k replied:
This is for readability of test code. I thought `removeLineBreak(newFile.contents)` is more readable than `newFile.contents.toString().replace(/\r?\n|\r/g, '')` but if you think this is unnecessary, I'm glad to restore the original code (and remove this).

rogeriopvl replied:
I mean, why do you wan't to remove the line break? `gulp-ejs` should not change the behaviour of `ejs` in anyway, don't you agree? Also that would be a breaking change for everyone using the plugin.

kt3k replied:
Sorry for confusing introduction of this helper. I will remove this.

The problem here is maybe that the expected value of `withpartial.ejs` is not 100% correct. So I'll try to fix that and remove this helper.

By the way, removing of line breaks starts at this commit (https://github.com/rogeriopvl/gulp-ejs/commit/8ff732f734087fca4e4fd95bb7eaf70bdd57d4ea). That was because of `ejs`'s breaking change and I think it's not a problem of this plugin.

kt3k commented:
I fixed the expected rendering file `test/expected/outputWithPartial.html` and removed `removeLineBreak` helper method.

The rendering result of `ejs` has been changed since this commit (https://github.com/rogeriopvl/gulp-ejs/commit/8ff732f734087fca4e4fd95bb7eaf70bdd57d4ea) and that's why I fixed the expected file. I think this change is not a problem here because it is caused by `ejs` itself, not this plugin.

rogeriopvl commented on test/expected/outputWithPartial.html:
```diff
@@ -1,3 +1,4 @@
 
-<h3><p>Hello rpvl!</p></h3>
+<h3><p>Hello rpvl!</p>
+</h3>
```
Why is this output now is expected with a line break here?

kt3k replied:
Because this is the actual rendering of `withpartial.ejs` by the current version of ejs which this module depends now.

Please see this commit (https://github.com/rogeriopvl/gulp-ejs/commit/8ff732f734087fca4e4fd95bb7eaf70bdd57d4ea). At that point, ejs rendering was changed and the workaround (of removing line breaks) was introduced, but I don't see any reason to keep that workaround. So I removed the workaround and fixed the fixture like the above.

I hope this makes sense.

rogeriopvl replied:
Makes sense! Thanks :+1:

rogeriopvl commented:
Released as `v2.2.0`
